docs/reference-guides/block-api/block-bindings.md

```diff
@@ -0,0 +1,296 @@
+- `attribute_name`: The name of the attribute.
+
+#### Server registration Core examples
+
+There are a few examples in Core that can be used as a reference.
+
+- Post Meta. [Source code](https://github.com/WordPress/wordpress-develop/blob/68d2421d758498095449006a055ad8ce5b7a8a40/src/wp-includes/block-bindings/post-meta.php#L59)
+- Pattern overrides. [Source code](https://github.com/WordPress/wordpress-develop/blob/68d2421d758498095449006a055ad8ce5b7a8a40/src/wp-includes/block-bindings/pattern-overrides.php#L36)
+- Twenty Twentyfive theme. [Source code](https://github.com/WordPress/wordpress-develop/blob/68d2421d758498095449006a055ad8ce5b7a8a40/src/wp-content/themes/twentytwentyfive/functions.php#L130)
+
+
+### Editor registration
+
+Editor registration on the client allows defining what the bound block will do when the value is retrieved or when the value is edited.
+
+The function to register a custom source is `registerBlockBindingsSource( args )`.
+
+- `args` is an `object` with the following structure:
+    - `name` is a `string` with the unique and machine-readable name.
+    - `label` is a `string` with the human readable name of the custom source. (optional)
+    - `usesContext` is an `array` with the block context that the custom source may need. (optional)
+    - `getValues` is a `function` that retrieves the values from the source. (optional)
+    - `setValues` is a `function` that allows updating the values connected to the source. (optional)
+    - `canUserEditValue` is a `function` to determine if the user can edit the value. The user won't be able to edit by default. (optional)
+    - `getFieldsList` is a private `function`. It cannot be used yet by third-party developers. It creates a list for the block bindings UI with post meta.
+
+The `label` argument will override the one defined on the server if they are different.
+
+
+This example will show a custom post meta date in the editor and, if it doesn't exist, it will show today's date. The user can edit the value of the date. (Caution: this example is not formatting the user input as a date, it's only for educational purposes.)
+
+```js
+import {
+	registerBlockBindingsSource,
+} from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+registerBlockBindingsSource( {
+	name: 'wpmovies/visualization-date',
+	label: __( 'Visualization Date' ),
+	useContext: [ 'postId', 'postType' ],
+	setValues( { select, dispatch, context, bindings } ) {
+		dispatch( coreDataStore ).editEntityRecord(
+			'postType',
+			context?.postType,
+			context?.postId,
+			{
+				meta: {
+					wp_movies_visualization_date: bindings?.content?.newValue,
+				},
+			}
+		);
+	},
+	getValues( { select, context } ) {
+		let wpModiesVisualizationDate;
+		const { getEditedEntityRecord } = select( coreDataStore );
+		if ( context?.postType && context?.postId ) {
+			return getEditedEntityRecord(
+				'postType',
+				context?.postType,
+				context?.postId
+			).meta?.wp_movies_visualization_date;
```

Comment:
I think you mean to set `wpMoviesVisualizationDate` rather than return here, otherwise the variable is always undefined.

Reply:
Yep. It should be 
```
wpMoviesVisualizationDate = getEditedEntityRecord(
				'postType',
				context?.postType,
				context?.postId
			).meta?.wp_movies_visualization_date;
```